cmake/modules/CheckYasm.cmake

```diff
@@ -10,14 +10,14 @@ macro(check_yasm_support _object_format _good_result _better_result)
       set(CMAKE_REQUIRED_QUIET true)
       include(CheckCXXSourceCompiles)
       check_cxx_source_compiles("
-      #if defined(__x86_64__) && defined(__ILP32__)
+      #if defined(_ILP32) || defined(__ILP32__)
```

Comment:
why? i think this check matches the expectation of x32 ABI.

Reply:
@tchaikov 
```
nstcc4@nstcloudcc4:~$ gcc -m32 -dM -E -xc - < /dev/null | grep -i 'x86_64\|ilp32'
#define __ILP32__ 1
#define _ILP32 1
nstcc4@nstcloudcc4:~$ gcc -m64 -dM -E -xc - < /dev/null | grep -i 'x86_64\|ilp32'
#define __x86_64 1
#define __x86_64__ 1
```